src/execution/execute.js

```diff
@@ -792,87 +792,54 @@ function completeValueCatchingError(
   path: ResponsePath,
   result: mixed,
 ): MaybePromise<mixed> {
-  // If the field type is non-nullable, then it is resolved without any
-  // protection from errors, however it still properly locates the error.
-  if (isNonNullType(returnType)) {
-    return completeValueWithLocatedError(
-      exeContext,
-      returnType,
-      fieldNodes,
-      info,
-      path,
-      result,
-    );
-  }
-
-  // Otherwise, error protection is applied, logging the error and resolving
-  // a null value for this field if one is encountered.
   try {
-    const completed = completeValueWithLocatedError(
-      exeContext,
-      returnType,
-      fieldNodes,
-      info,
-      path,
-      result,
-    );
+    let completed;
+    if (isPromise(result)) {
```

Comment:
`completeValue` only called in 3 places:

https://github.com/graphql/graphql-js/blob/43ff3e35c699d1b6260126465b84810678d4543e/src/execution/execute.js#L907-L912

I moved this code to `completeValueCatchingError` so nothing changed here.

https://github.com/graphql/graphql-js/blob/43ff3e35c699d1b6260126465b84810678d4543e/src/execution/execute.js#L919-L929

This code is called after promise uplift so nothing changed here.

https://github.com/graphql/graphql-js/blob/43ff3e35c699d1b6260126465b84810678d4543e/src/execution/execute.js#L849

Also nothing is broken here since this call moved to `completeValueCatchingError` and `completeValueCatchingError` now handles promise uplift by itself.